README.md

```diff
@@ -1,39 +1,132 @@
 
-* **Culture** - Paulo, one of your future colleagues, wrote about it. [https://medium.com/we-are-swat/what-its-like-to-work-on-the-swat-team-30d3a081e116]
+Hint: Note that the score in % of the two posts are equal, but the real "values" are significantly differents
 
-* **You** - by joining our team, feel free to question all the items below and propose new ideas on how we work. This is definitely not a static thing.
+You should be doing the solution on a specific branch. Once you have something to delivery, you can open a Pull Request. You can use the Pull Request if you'd like some feedback on your code or to discuss something with us.
 
-* **GitHub** - all our code is hosted here. We use Pull Requests and do code reviews for those. Everyone on the team reviews PRs. It's expected that you write quality code with automated tests. Once a PR is reviewed and accepted, the person who opened the PR should merge it and delete the branch.
+### Things We value
 
-* **Continuous Integration** - we use Semaphore for our CI needs. Everytime we push to a branch, our test suite runs on Semaphore.
+- Clean code, we want you to show us the best code you can do
+- If you are familiar with TDD, BDD or any testing process, please show us your skills
 
-* **Slack** - we mostly communicate asynchronously between the development team and with other teams. This is our chat tool. Abuse it.
+## Solution
 
-* **Pivotal Tracker** - our products' backlog is managed in Pivotal Tracker. Once we estimate all known user-stories, the sprint backlog is automatically built. Each team member will select a user-story to work from that sprint backlog and will update the needed tasks' status. This way we keep focused until the end of the sprint.
+My solution is a Node.js service exposing an API on port `8080` with the following endpoints:
+- `GET /posts` 
+    - returns a list of posts in the expected order in JSON format
+- `GET /posts/:post_id` 
+    - returns the post with the specified ID
+    - returns error if post with specified ID is not found
+- `POST /posts` 
+    - create new post
+    - expects parameter `title` in body
+    - returns the newly created post
+- `PUT /upvote/:post_id` or `PUT /posts/:post_id/upvote`
+    - upvote specified post
+- `PUT /downvote/:post_id` or `PUT /posts/:post_id/downvote`
+    - downvote specified post
 
-* **Support tasks** - We are paranoid about providing top notch support to our customers. We have a dedicated customer support team working full-time communicating with clients. Our development team works very closely with the customer support team delivering happiness to our clients.
+### Technologies
+
+The following are the main technologies used:
+- Node.js
+- TypeScript
+- Express
+- Joi
+- Knex
+- Jest
+- PostgreSQL
```

Comment:
Why did you choose PostgreSQL?

Imagine that we want to deploy your solution using Docker. So running in multiple containers. Would you still choose PostgreSQL? Which aspects do you think are important while choosing a database for this type of architecture?

Reply:
> Why did you choose PostgreSQL?

I chose PostgreSQL because felt like a RBMS would be a good fit for this project and PostgreSQL happens to be one I'm confortable with. I considered using some kind of NoSQL DB but when I decided to have a list of votes per user and per post I thought a relational DB would be a better fit.

> Imagine that we want to deploy your solution using Docker. So running in multiple containers. Would you still choose PostgreSQL? Which aspects do you think are important while choosing a database for this type of architecture?

I'm no expert on Docker or DBs, so I'm unaware of the possible pitfalls of using a PostgreSQL in multiple containers in production. In general I feel like a good idea for such deployments would be choose a DB system which can be easily clustered, so you can use replication techniques like master/slave. What kind of important aspects or pitfalls of this approach am I not considering?

I used Docker because it allows me to have consistent dev environment. One thing I would definitely change when adapting my solution to a production environment would be to add a volume to the DB container so the DB itself could be persisted. As it stands the data is lost if the container is deleted.